lib/matplotlib/rcsetup.py

```diff
@@ -32,20 +33,34 @@
 from cycler import Cycler, cycler as ccycler
 
 
-# The capitalized forms are needed for ipython at present; this may
-# change for later versions.
-interactive_bk = [
-    'GTK3Agg', 'GTK3Cairo', 'GTK4Agg', 'GTK4Cairo',
-    'MacOSX',
-    'nbAgg',
-    'QtAgg', 'QtCairo', 'Qt5Agg', 'Qt5Cairo',
-    'TkAgg', 'TkCairo',
-    'WebAgg',
-    'WX', 'WXAgg', 'WXCairo',
-]
-non_interactive_bk = ['agg', 'cairo',
-                      'pdf', 'pgf', 'ps', 'svg', 'template']
-all_backends = interactive_bk + non_interactive_bk
+# Deprecation of module-level attributes using PEP 562
+_deprecated_interactive_bk = backend_registry.list_builtin(BackendFilter.INTERACTIVE)
+_deprecated_non_interactive_bk = backend_registry.list_builtin(
+    BackendFilter.NON_INTERACTIVE)
+_deprecated_all_backends = backend_registry.list_builtin()
+
+_deprecated_names_and_args = {
+    "interactive_bk": "matplotlib.backends.registry.BackendFilter.INTERACTIVE",
+    "non_interactive_bk": "matplotlib.backends.registry.BackendFilter.NON_INTERACTIVE",
+    "all_backends": "",
+}
+
+
+def __getattr__(name):
```

Comment:
This should use `_api.caching_module_getattr` with suitable `_api.deprecated ` decorators.

Reply:
OK, I have switched to using `_api.caching_module_getattr`. I have kept the separate private attributes such as `_deprecated_interactive_bk` but these could just be moved to the individual deprecated properties if that is preferred.

Reply:
It doesn't look like they're used elsewhere, so they probably could be inlined. The lambdas could be changed to decorated methods if that helps readability.

Reply:
I have inlined them and changed to decorated methods as suggested, thanks.